### ISSUE TITLE
Key error 'classifer' #27

### ISSUE THREAD

**Issue by yashmehta10** [globality-corp/sklearn-hierarchical-classification#52]
Connecting the old issue. #27 
I'm facing the same issue. I am using SVM as the classifier.  
![image](https://user-images.githubusercontent.com/17799552/120126520-e05dd880-c1ff-11eb-8751-c572be18dcec.png)

The fit is completed but some nodes are terminated due to lack of data, can this be the reason?
![image](https://user-images.githubusercontent.com/17799552/120126384-6decf880-c1ff-11eb-94ee-37a59e7934b8.png)

Unfortunately I won't be able to provide the data. 

Thank you.

**Comment by Meresmata** [comment]
Hi,

I also ran into a the Key Error classifier! I am currently using it with in a jupyternotebook in Pycharm. I additionally get a text: _train_local_classifier() - not enough training data available to train, classification in branch will terminate at node <ROOT>

For every parent node in the hierachy. 

I use a test set of about 1.2 M samples and 1.8 M samples in the Trainingset. The think is: there should be enough DATA!  The base estimator I use is a tree.

Data descripion:
y: int as in the class hierarchy but with str-> int
X: float [0...1], float[0...1], int,  the floats are relative intensities and the integer the week of the year.

Code:
``` python
from sklearn_hierarchical_classification.classifier import HierarchicalClassifier
from sklearn_hierarchical_classification.constants import ROOT
from sklearn_hierarchical_classification.metrics import h_fbeta_score, multi_labeled

class_hierarchy = {
  ROOT: ["16", "32", "48"],
  "16": ["20", "24", "29"],
  "32": ["37", "41"],
  "48": ["52", "56"],
  "20": ["21", "22", "23"],
  "24": ["25", "26", "27"],
  "52": ["53", "54"],
  "56": ["57", "58"],


}
hiera_clf = HierarchicalClassifier(
        base_estimator=DecisionTreeClassifier(max_depth=13, ccp_alpha=1e-6,),
        class_hierarchy=class_hierarchy,
    )
hiera_clf.fit(train_dropped, y_train)

y_pred = hiera_clf.predict(test_dropped)
with multi_labeled(y_test, y_pred, hiera_clf.graph_) as (y_test_, y_pred_, graph_):
    h_fbeta = h_fbeta_score(
        y_test_,
        y_pred_,
        graph_,
    )
    print("h_fbeta_score: ", h_fbeta)
```

**Comment by Meresmata** [comment]
So, I am currently digging into the problem:

After fitting, in the prediction step. When the _recursive_predict mehtod is called... these keys: exist: X and metafeatures. With metafeatures: n_samples and n_targets are zero! There is no reference for the classifier!

**Comment by Meresmata** [comment]
Prediction worked when I transformed the y from int to str.
Now, got an error in calculation of h_fbeta_score. In the function h_precision_score: I got 0/0.  (module metrics, line 153.)

But at least I got a prediction... which had I had to convert str->int. Which worked, after I found out, that the prediction is a masked array. (`y_pred.data(astype(int)`).
